lib/ansible/modules/apt.py

```diff
@@ -634,8 +634,11 @@ def expand_pkgspec_from_fnmatches(m, pkgspec, cache):
 
             pkgname_pattern, version_cmp, version = package_split(pkgspec_pattern)
 
+            if pkgname_pattern.startswith(('?', '~', '(?', '(~')):
```

Comment:
@jmechnich Do you mind adding an integration test for this change? I am not quite sure if I understand how this change works.

Reply:
@Akasurde I cannot really think of a way to test this without actually calling apt-get or similar in a appropriately set up OS. Right now, the tests emulate certain operations using an artificial package cache.
The general idea of this patch is to exclude package names starting with `?` or `~` from normal processing as they are not *legal* patterns for i.e. apt-cache operations.
The issue with testing this in the realm of ansible is mainly that the effect of those patterns is determined on each target host, individually.